app/uk/gov/hmrc/gform/gformbackend/GformConnector.scala

```diff
@@ -119,12 +120,29 @@ class GformConnector(ws: WSHttp, baseUrl: String) {
             }
 
           }
-          .recover {
+          .recoverWith {
             case UpstreamErrorResponse.WithStatusCode(statusCode, _) if statusCode == StatusCodes.NotFound.intValue =>
               logger.info(
                 s"Attempt to access form $formIdData, but form not found in MongoDB, attempt to look for $formIdDataOriginal as a fallback failed."
               )
-              None
+
+              def getFormByLegacyIds(legacyIds: NonEmptyList[FormTemplateId]): Future[Option[Form]] =
+                legacyIds match {
+                  case NonEmptyList(legacyFormId, tail) =>
```

Comment:
To eliminate one level of nesting this can be simplified to 
```
val NonEmptyList(legacyFormId, tail) = legacyIds
```
(feel free to ignore this)